gen3-integration-tests/gen3_ci/jenkins_jobs/check-kube-pod.groovy

```diff
@@ -0,0 +1,97 @@
+/*
+    String parameter JOBNAME
+    eg - manifest-indexing
+
+    String parameter LABELNAME
+    eg - sowerjob
+
+    String parameter EXPECTFAILURE
+    default value - False
+*/
+pipeline {
+    agent {
+        node {
+            label 'gen3-ci-worker'
+        }
+    }
+    stages {
+        stage('Clean old workspace') {
+            steps {
+                cleanWs()
+            }
+        }
+        stage('Initial Setup') {
+            steps {
+                // cloud-automation
+                checkout([
+                  $class: 'GitSCM',
+                  branches: [[name: 'refs/heads/master']],
+                  doGenerateSubmoduleConfigurations: false,
+                  extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'cloud-automation']],
+                  submoduleCfg: [],
+                  userRemoteConfigs: [[credentialsId: 'PlanXCyborgUserJenkins', url: 'https://github.com/uc-cdis/cloud-automation.git']]
+                ])
+            }
+        }
+        stage('Check Kube Pod') {
+            options {
+                    timeout(time: 5, unit: 'MINUTES')   // timeout on this stage
+                }
+            steps {
+                dir("check-kube-pod"){
+                    script {
+                        try {
+                            sh '''#!/bin/bash +x
+                            set -e
+                            export GEN3_HOME=\$WORKSPACE/cloud-automation
+                            export KUBECTL_NAMESPACE=\${NAMESPACE}
+                            source $GEN3_HOME/gen3/gen3setup.sh
+
+                            N_ATTEMPTS=10
+                            echo "NAMESPACE : $KUBECTL_NAMESPACE"
+                            for (( i=1; i<=N_ATTEMPTS; i++ )); do
+                                sleep 10
+                                echo "Waiting for $JOBNAME job pod ..."
+
+                                # checking if there are pods with labelname mentioned in parameters
+                                podNames=$(g3kubectl -n $KUBECTL_NAMESPACE get pod --sort-by=.metadata.creationTimestamp -l app=$LABELNAME -o jsonpath="{.items[*].metadata.name}")
```

Comment:
don't we need to also filter by the job name? this might look at all sower jobs

Reply:
Yes I think we need a filter here. If they are multiple pods with same app label, it will pick up the latest pod. But not the latest pod of the job we want for the script